lib/sisimai/bite/email/courier.rb

```diff
@@ -0,0 +1,265 @@
+      #                                   part or nil if it failed to parse or
+      #                                   the arguments are missing
+      def scan(mhead, mbody)
+        return nil unless mhead
+        return nil unless mbody
+
+        match  = 0
+        match += 1 if mhead['from']    =~ Re0[:from]
+        match += 1 if mhead['subject'] =~ Re0[:subject]
+        if mhead['message-id']
+          # Message-ID: <courier.4D025E3A.00001792@5jo.example.org>
+          match += 1 if mhead['message-id'] =~ Re0[:'message-id']
+        end
+        return nil if match.zero?
+
+        dscontents = [Sisimai::Bite.DELIVERYSTATUS]
+        hasdivided = mbody.split("\n")
+        havepassed = ['']
+        rfc822list = []     # (Array) Each line in message/rfc822 part string
+        blanklines = 0      # (Integer) The number of blank lines
+        readcursor = 0      # (Integer) Points the current cursor position
+        recipients = 0      # (Integer) The number of 'Final-Recipient' header
+        commandtxt = ''     # (String) SMTP Command name begin with the string '>>>'
+        connvalues = 0      # (Integer) Flag, 1 if all the value of $connheader have been set
+        connheader = {
+          'date'  => '',    # The value of Arrival-Date header
+          'rhost' => '',    # The value of Reporting-MTA header
+          'lhost' => '',    # The value of Received-From-MTA header
+        }
+        v = nil
+
+        hasdivided.each do |e|
+          # Save the current line for the next loop
+          havepassed << e
+          p = havepassed[-2]
+
+          if readcursor.zero?
+            # Beginning of the bounce message or delivery status part
+            if e =~ Re1[:begin]
+              readcursor |= Indicators[:deliverystatus]
+              next
+            end
+          end
+
+          if readcursor & Indicators[:'message-rfc822'] == 0
+            # Beginning of the original message part
+            if e =~ Re1[:rfc822]
+              readcursor |= Indicators[:'message-rfc822']
+              next
+            end
+          end
+
+          if readcursor & Indicators[:'message-rfc822'] > 0
+            # After "message/rfc822"
+            if e.empty?
+              blanklines += 1
+              break if blanklines > 1
+              next
+            end
+            rfc822list << e
+
+          else
+            # Before "message/rfc822"
+            next if readcursor & Indicators[:deliverystatus] == 0
+            next if e.empty?
+
+            if connvalues == connheader.keys.size
+              # Final-Recipient: rfc822; kijitora@example.co.jp
+              # Action: failed
+              # Status: 5.0.0
+              # Remote-MTA: dns; mx.example.co.jp [192.0.2.95]
+              # Diagnostic-Code: smtp; 550 5.1.1 <kijitora@example.co.jp>... User Unknown
+              v = dscontents[-1]
+
+              if cv = e.match(/\A[Ff]inal-[Rr]ecipient:[ ]*(?:RFC|rfc)822;[ ]*([^ ]+)\z/)
```

Comment:
__[rubocop]__
* Style/NumericPredicate: Use `(readcursor & Indicators[:deliverystatus]).zero?` instead of `readcursor & Indicators[:deliverystatus] == 0`. [:ref](https://github.com/bbatsov/ruby-style-guide#predicate-methods)

<!-- Comment by [SideCI](https://sideci.com) -->